### PR TITLE
Langchain vectorstore integration with Kinetica

### PR DESCRIPTION
**Pull request by am-kinetica**
- **Description:** New vectorstore integration with the Kinetica database 
  - **Issue:** 
  - **Dependencies:** the Kinetica Python API `pip install gpudb==7.2.0.0b`,
  - **Tag maintainer:** @baskaryan,
  - **Twitter handle:**

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
please dont update anything in langchain, we should NOT reexport those

**Review by am-kinetica — COMMENTED** [review]
@hwchase17 - Could you please point out where this reexport (as you mentioned) has been done. Maybe I had to do it since some step was failing without it. Since I am new to this process, I might have done something which was wrong.

**Comment by am-kinetica** [comment]
Updated from master

**Comment by am-kinetica** [comment]
@efriis - Could you please comment on what needs to be done on this PR ?

File: libs/langchain/langchain/vectorstores/kinetica.py
Comment (efriis) [inline libs/langchain/langchain/vectorstores/kinetica.py:1]: revert all changes in libs/langchain

File: poetry.lock
Comment (efriis) [inline poetry.lock:1]: revert

File: libs/community/langchain_community/vectorstores/kinetica.py [hunk 1]
```diff
@@ -0,0 +1,904 @@
...
+import numpy as np
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import BaseSettings
+from langchain_core.vectorstores import VectorStore
+
+from langchain_community.vectorstores.utils import maximal_marginal_relevance
+
+
+class DistanceStrategy(str, enum.Enum):
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/kinetica.py:23]: Can you reuse `langchain_community.vectorstores.utils.DistanceStrategy` instead?

**Review by efriis — COMMENTED** [review]
Hey there! Are all these methods in the vectorstore implementation necessary? There are a few nonstandard things going on here that will make this vectorstore difficult to use as well:

- assumed dimensionality of openai embeddings -> does that need to be set on initialization?
- documentation is missing - also how to run integration tests / set up a demo instance

I'm going to close this PR for now, and if you add these, feel free to reopen! The lighter weight the implementation is here, the easier it will be to review.

**Comment by am-kinetica** [comment]
@efriis - Could you please comment on what needs to be done on this PR ?

> Hey there! Are all these methods in the vectorstore implementation necessary? There are a few nonstandard things going on here that will make this vectorstore difficult to use as well:
> 
>     * assumed dimensionality of openai embeddings -> does that need to be set on initialization?
> 
>     * documentation is missing - also how to run integration tests / set up a demo instance
> 
> 
> I'm going to close this PR for now, and if you add these, feel free to reopen! The lighter weight the implementation is here, the easier it will be to review.

I have followed pretty much the way `pgvector` vector store implementation was done. Could you please be more elaborate on what is non-standard here ? I am sorry but I am doing this the first time and not sure which one to follow as a reference.

File: libs/langchain/langchain/vectorstores/kinetica.py
Comment (am-kinetica) [reply libs/langchain/langchain/vectorstores/kinetica.py:1]: Here also, I actually followed the what was done for `pgvector`. So, the question is if I revert these changes (remove the file from here), won't there be build issues ?

**Comment by efriis** [comment]
There are differences between how things were implemented before the community split and now - sorry for the confusion!

All of these re-exports from langchain are only for backwards compatibility. New implementations should only exist in community.

**Comment by efriis** [comment]
https://github.com/langchain-ai/langchain/discussions/14243

**Comment by am-kinetica** [comment]
> There are differences between how things were implemented before the community split and now - sorry for the confusion!
> 
> All of these re-exports from langchain are only for backwards compatibility. New implementations should only exist in community.

Thanks for the clarification, will change accordingly.

File: libs/community/langchain_community/vectorstores/kinetica.py (hunk 1 above)
Comment (am-kinetica) [reply libs/community/langchain_community/vectorstores/kinetica.py:23]: The current `pgvector` vector store implementation has this:

**Comment by am-kinetica** [comment]
Modified the vectorstore implementation

**Comment by am-kinetica** [comment]
I don't think I am allowed to reopen this PR. 
@efriis @hwchase17 Do you think you could open it or do I create a new PR ?

**Review by am-kinetica — COMMENTED** [review]
Removed everything from `langchain` and added notebook for Kinetica vectorstore integration

File: poetry.lock
Comment (baskaryan) [inline poetry.lock:1]: can we undo this change

File: libs/langchain/poetry.lock
Comment (baskaryan) [inline libs/langchain/poetry.lock:1]: can we undo this change

File: poetry.lock
Comment (am-kinetica) [reply poetry.lock:1]: Removed

File: libs/langchain/poetry.lock
Comment (am-kinetica) [reply libs/langchain/poetry.lock:1]: Removed

Comment (am-kinetica) [reply libs/langchain/poetry.lock:1]: I am getting this error now
`Run poetry lock --check
Creating virtualenv langchain-monorepo in /home/runner/work/langchain/langchain/.venv
poetry lock --check is deprecated, use `poetry check --lock` instead.
Error: poetry.lock is not consistent with pyproject.toml. Run `poetry lock [--no-update]` to fix it.
Error: Process completed with exit code 1.
`

**Comment by am-kinetica** [comment]
Created a new PR - https://github.com/langchain-ai/langchain/pull/18102
Closing this PR.
